zebrad/src/application.rs

```diff
@@ -1,27 +1,24 @@
 //! Zebrad Abscissa Application
```

Comment:
This PR seems to have memory safety issues, or possibly this is another instance of bug #6812.

>   process didn't exit successfully: `/home/runner/work/zebra/zebra/target/release/deps/zebrad-ec601d4e69bdf498` (signal: 11, SIGSEGV: invalid memory reference)

https://github.com/ZcashFoundation/zebra/actions/runs/5151819470/jobs/9277360620?pr=6801#step:15:3194

Reply:
We've confirmed that bug #6812 only happens on macOS x86_64 with Rust 1.70, so a Linux bug on Rust 1.71-beta seems like a different issue.

How did you want to move forward with this?

Reply:
> How did you want to move forward with this?

I think it happened in `args_with_subcommand_pass_through()`, I'm hoping it'll happen again for the most recent commit.

Reply:
This has been fixed by making sure that the index is always within the vector.